src/core/ack_tracker.c

```diff
@@ -283,11 +336,12 @@ QuicAckTrackerOnAckFrameAcked(
 
     //
     // Drop all packet numbers less than or equal to the largest acknowledged
-    // packet number.
+    // packet number - Reordering Threshold + 1. This is so that we dont lose 
+    // memory of packets that are missing and are yet to be reported. 
     //
     QuicRangeSetMin(
         &Tracker->PacketNumbersToAck,
-        LargestAckedPacketNumber + 1);
+        LargestAckedPacketNumber - Connection->ReorderingThreshold + 2);
```

Comment:
I thought we decided this wouldn't help, without actually tracking the first gap before.

Reply:
Oh yeah, it does help us in few edge cases but not always.